ui/src/pipeline/pipeline.consumer.pact.test.ts

```diff
@@ -179,31 +182,36 @@ pactWith(options, provider => {
             state: `pipelines with datasource id ${id} do not exist`,
             uponReceiving: getByDatasourceIdRequestTitle(id),
             withRequest: getByDatasourceIdRequest(id),
-            willRespondWith: notFoundResponse,
+            willRespondWith: getByDatasourceIdEmptyResponse,
           });
         });
 
-        it('throws an error', async () => {
-          await expect(
-            restService.getPipelineByDatasourceId(id),
-          ).rejects.toThrow(Error);
+        it('returns an empty pipeline array', async () => {
+          const pipelines = await restService.getPipelineByDatasourceId(id);
+
+          expect(pipelines).toStrictEqual([]);
         });
       });
 
       describe('with NaN as requested datasource id', () => {
         beforeEach(async () => {
           await provider.addInteraction({
-            state: 'any state',
+            state: 'pipelines with datasource id NaN return all pipelines',
```

Comment:
The state of the provider can be arbitrary for this tested interaction, i.e. it does not matter whether any pipelines exist or not on the provider side.
When sending `NaN` as datasource id, the response should always be an empty array because, as far as I know, there can't be pipelines with a datasource id of `NaN`.

```suggestion
            state: 'any state',
```

Reply:
You are right that there will be no pipelines with datasourceId `NaN`, but the backend api returns all pipelines, in case the datasourceId param is `NaN`
https://github.com/jvalue/ods/blob/acda73c1fb05c7f5204463d9dc150dd1d4b93975/pipeline/src/api/rest/pipelineConfigEndpoint.ts#L101-L118

Reply:
Oh you are right, I missed that. Intuitively I would have expected an empty array or a bad request status as response in that case. @georg-schwarz is there a reason for that behavior?

Reply:
I agree. Either `[]` or 400 makes sense.

Reply:
@georg-schwarz I guess you mean 404 (because we have not found any for given datasourceId)?

Reply:
@felix-oq The changes are pushed and due to the API change I have applied all of your pact changes